benchmarks/Benchmarks.Trace/AspNetCoreBenchmark.cs

```diff
@@ -0,0 +1,141 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Benchmarks.Trace
+{
+    [MemoryDiagnoser]
+    public class AspNetCoreBenchmark
+    {
+        private static readonly HttpClient Client;
+
+        static AspNetCoreBenchmark()
+        {
+            Environment.SetEnvironmentVariable(ConfigurationKeys.TraceEnabled, "0");
+            Environment.SetEnvironmentVariable(ConfigurationKeys.StartupDiagnosticLogEnabled, "0");
+
+            var settings = new TracerSettings
+            {
+                StartupDiagnosticLogEnabled = false,
+                DiagnosticSourceEnabled = true
+            };
+
+            Tracer.Instance = new Tracer(settings, new DummyAgentWriter(), null, null, null);
+
+            var builder = new WebHostBuilder()
+                .UseStartup<Startup>();
+
+            var testServer = new TestServer(builder);
+            Client = testServer.CreateClient();
+
+            Datadog.Trace.ClrProfiler.Instrumentation.Initialize();
+
+            HomeController.Initialize();
+        }
+
+        [Benchmark]
+        public Task<string> SendRequest()
+        {
+            return Client.GetStringAsync("/Home");
+        }
+
+        private class Startup
+        {
+            public void ConfigureServices(IServiceCollection services)
+            {
+                services.AddMvc();
+            }
+
+            public void Configure(IApplicationBuilder builder)
+            {
+                builder.UseMvcWithDefaultRoute();
+            }
+        }
+    }
+
+    /// <summary>
+    /// Simple controller used for the aspnetcore benchmark
+    /// </summary>
+    public class HomeController : Controller
+    {
+        private static readonly HttpRequestMessage HttpRequest = new HttpRequestMessage();
+        private static readonly HttpMessageHandler Handler = new CustomHttpMessageHandler();
+        private static readonly object BoxedCancellationToken = new CancellationToken();
+        private static int _mdToken;
+        private static IntPtr _guidPtr;
+
+        internal static void Initialize()
+        {
+            HttpMessageHandlerIntegration.HttpClientHandler = typeof(CustomHttpMessageHandler).FullName;
+
+            var methodInfo = typeof(HttpMessageHandler).GetMethod("SendAsync", System.Reflection.BindingFlags.NonPublic | System.Reflection.BindingFlags.Instance);
+
+            _mdToken = methodInfo.MetadataToken;
+            var guid = typeof(HttpMessageHandler).Module.ModuleVersionId;
+
+            _guidPtr = Marshal.AllocHGlobal(Marshal.SizeOf(guid));
+
+            Marshal.StructureToPtr(guid, _guidPtr, false);
+        }
+
+        public async Task<string> Index()
+        {
+            var task = (Task)HttpMessageHandlerIntegration.HttpMessageHandler_SendAsync(
+                Handler,
+                HttpRequest,
+                BoxedCancellationToken,
+                111,
```

Comment:
Are you able to reference `Datadog.Trace.ClrProfiler.Emit.OpcodeValue.Callvirt` instead of 111? Or maybe just adding a clarifying comment here that the value indicates a CallVirt instruction would be helpful

Reply:
Good point